### ISSUE TITLE
TypeError: "NoneType"  when running count_coverage

### ISSUE THREAD

**Issue by nickschurch** [pysam-developers/pysam#603]
When I load a bam file and call `count_coverage` on a region I'm getting an none error:

```
> thisbam = pysam.AlignmentFile(bam1,"rb")
> thisbam.text
'@HD\tVN:1.3\tSO:coordinate\n@SQ\tSN:1\tLN:30427671\n@SQ\tSN:2\tLN:19698289\n@SQ\tSN:3\tLN:23459830\n@SQ\tSN:4\tLN:18585056\n@SQ\tSN:5\tLN:26975502\n@SQ\tSN:Mt\tLN:366924\n@SQ\tSN:Pt\tLN:154478\n'
> thisbam.count("1",10,100000)
1724
> thisbam.count_coverage("1",10,100000)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-44-5c08c453e230> in <module>()
----> 1 thisbam.count_coverage("1",10,100000)

pysam/libcalignmentfile.pyx in pysam.libcalignmentfile.AlignmentFile.count_coverage()

TypeError: 'NoneType' object has no attribute '__getitem__'
```
Python 2.7.13 running pysam 0.13 on centos 6.3

**Comment by AndreasHeger** [comment]
Thanks. Is it possible to share some sample data where this error occurs - I have trouble reproducing it.

**Comment by mv911119** [comment]
[af.zip](https://github.com/pysam-developers/pysam/files/1699163/af.zip)
Here I have the same problem and this is my 16S data.
the chromosomes' names are 0,1,2,3,4.....
And Nick used Python 2 which could be the problem.
But I am using Python 3.6.2 and Centos Linux 7 And having the same in chromosome region reading

import pysam
sbm1 = pysam.AlignmentFile("/afsecondfiltered.bam","rb")
sbm1.count_coverage(contig="1", start= 1, stop=12)
Traceback (most recent call last):

  File "<ipython-input-68-2e971b60de51>", line 3, in <module>
    sbm1.count_coverage(contig="1", start= 1, stop=12)

  File "pysam/libcalignmentfile.pyx", line 1235, in pysam.libcalignmentfile.AlignmentFile.count_coverage

TypeError: 'NoneType' object is not subscriptable

**Comment by zhoudreames** [comment]
I also get this problem ,do you know how to solve it ? thanks
Traceback (most recent call last):
  File "/home/Mpzhang/Mzhou/Biosoft/NucFreq-master/NucPlot.py", line 177, in <module>
    cov = getCovByBase(contig, start, end)
  File "/home/Mpzhang/Mzhou/Biosoft/NucFreq-master/NucPlot.py", line 154, in getCovByBase
    coverage = bam.count_coverage(
  File "pysam/libcalignmentfile.pyx", line 1586, in pysam.libcalignmentfile.AlignmentFile.count_coverage
TypeError: 'NoneType' object is not subscriptable

**Comment by zhoudreames** [comment]
@AndreasHeger could you help me ?

**Comment by wckdouglas** [comment]
@zhoudreames Do you know what pysam version are you on?

**Comment by zhoudreames** [comment]
> @zhoudreames Do you know what pysam version are you on?
thanks for your reply,my pysam version is 0.16.0.1
![image](https://user-images.githubusercontent.com/54714639/120053105-01d18f80-c05b-11eb-8fef-b3e0cccca4b9.png)

**Comment by wckdouglas** [comment]
umm interesting, can you share the bam file for reproducing this error?

**Comment by jmarshall** [comment]
Line 1586 of that file in 0.16.0.1 is `if seq[qpos] == 'A':` so I suspect this is similar to the previous problem with `read.query_qualities` being None but this time `read.seq` is None. This happens for reads with a `SEQ` field of `*`, which may be occasionally output by some aligners in some modes.

Likely the first issue was fixed by PR #621 and this one will have been fixed by 34fbe328198038ab2f27dd2b0df91b431350a1ea.
